internal/app/secretless/providers/kubernetessecrets/client.go

```diff
@@ -1,29 +1,38 @@
 package kubernetessecrets
 
 import (
-	"k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
-// NewKubeClient creates a Kubernetes API client using in-cluster config
-func NewKubeClient() (kubernetes.Interface, error) {
+// NewSecretsClient creates a SecretsClient for working with Secrets
+// in the pod namespace.
+// A SecretsClient uses an underlying Kubernetes API client
+// initialised with in-cluster config.
```

Comment:
Despite this being a simple file, I'd like to see these comments expanded.  Specifically, 

1. What is an "in-cluster config"
2. An ~2-3 sentence overview of what the code does, explained assuming no familiarity with the underlying concepts.

I'd really like to start writing all new code with the POV of "this should be completely understandable to a new dev with no domain knowledge."  And familiarity with kubernetes secrets shouldn't be assumed for a dev on secretless.